src/Components/ImagineControl.jsx

```diff
@@ -0,0 +1,165 @@
+import CopyIcon from '../assets/icons/Copy.svg'
+import {Helmet} from 'react-helmet-async'
+import AutoFixHighIcon from '@mui/icons-material/AutoFixHigh'
+import ContentCopyIcon from '@mui/icons-material/ContentCopy'
+import InputAdornment from '@mui/material/InputAdornment'
+import IconButton from '@mui/material/IconButton'
+
+
+/**
+ * This button hosts the ShareDialog component and toggles it open and
+ * closed.
+ *
+ * @return {React.ReactElement} The button react component, with a hosted
+ *   ShareDialog component
+ */
+export default function ShareControl() {
+  const [isDialogDisplayed, setIsDialogDisplayed] = useState(false)
+  const openedDialog = !!isDialogDisplayed
+
+
+  return (
+    <ControlButton
+      title='AI rendering'
+      icon={<AutoFixHighIcon className='icon-share' color='secondary'/>}
+      isDialogDisplayed={openedDialog}
+      setIsDialogDisplayed={setIsDialogDisplayed}
+      dialog={
+        <ShareDialog
+          isDialogDisplayed={openedDialog}
+          setIsDialogDisplayed={setIsDialogDisplayed}
+        />
+      }
+    />
+  )
+}
+
+
+/**
+ * The ShareDialog component lets the user control what state is
+ * included in the shared URL and assists in copying the URL to
+ * clipboard.
+ *
+ * @param {boolean} isDialogDisplayed
+ * @param {Function} setIsDialogDisplayed
+ * @return {React.Component} The react component
+ */
+function ShareDialog({isDialogDisplayed, setIsDialogDisplayed}) {
+  const cameraControls = useStore((state) => state.cameraControls)
+  const viewer = useStore((state) => state.viewer)
+  const model = useStore((state) => state.model)
+  const urlTextFieldRef = createRef()
+
+
+  useEffect(() => {
+    if (viewer) {
+      addCameraUrlParams(cameraControls)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [viewer, model])
+
+
+  const closeDialog = () => {
+    setIsDialogDisplayed(false)
+  }
+
+  const onCopy = async (event) => {
+    await navigator.clipboard.writeText(window.location.href)
+    window.open('https://discord.com/channels/853953158560743424/1126526910495740005', '_blank')
+    urlTextFieldRef.current.select()
+    closeDialog()
+  }
+
+  const handleCopyClick = async () => {
+    await navigator.clipboard.writeText(window.location.href)
+    // You can add some state or UI feedback to show that the link has been copied
+  }
+
+
+  return (
+    <Dialog
+      icon={<AutoFixHighIcon className='icon-share'/>}
+      headerText='AI Rendering'
+      isDialogDisplayed={isDialogDisplayed}
+      setIsDialogDisplayed={closeDialog}
+      actionTitle='Access the bot'
+      actionIcon={<CopyIcon className='icon-share'/>}
+      actionCb={onCopy}
+      content={
+        <Stack
+          alignItems='center'
+          justifyContent='flex-start'
+          spacing={0}
+          sx={{
+            width: '266px',
+            marginBottom: '4px',
+          }}
+        >
+          <Helmet>
+            <title>AI Rendering — BLDRS</title>
```

Comment:
Bldr Bot Rendering

and maybe no BLDRS?  I'm looking at my tabs..
- Github uses repo/issue, and doesn't include their name
- Google uses doc/mail subject and doesn't include their or app name
- Discord uses their name as prefix, after channel count
- Flux as a suffix after project name